client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -539,18 +539,29 @@ public PartitionLocation registerMapPartitionTask(
                     conf.clientRpcRegisterShuffleAskTimeout(),
                     ClassTag$.MODULE$.apply(PbRegisterShuffleResponse.class)));
 
-    if (partitionLocationMap == null) {
-      throw new CelebornIOException("Register shuffle failed for shuffle " + shuffleId);
-    }
-
     return partitionLocationMap.get(partitionId);
   }
 
   @Override
   public ConcurrentHashMap<Integer, PartitionLocation> getPartitionLocation(
-      int shuffleId, int numMappers, int numPartitions) {
-    return reducePartitionMap.computeIfAbsent(
-        shuffleId, (id) -> registerShuffle(shuffleId, numMappers, numPartitions));
+      int shuffleId, int numMappers, int numPartitions) throws CelebornIOException {
+    try {
+      return reducePartitionMap.computeIfAbsent(
+          shuffleId,
+          (id) -> {
+            try {
+              return registerShuffle(shuffleId, numMappers, numPartitions);
+            } catch (CelebornIOException e) {
+              throw new RuntimeException(e);
```

Comment:
Why not directly throw `e`?

Reply:
@SteNicholas  CelebornIOException is a checked exception，but in java.util.Map#computeIfAbsent(K key, Function<? super K, ? extends V> mappingFunction), the mappingFunction can not throw a checked exception